crates/sp-domains/src/core_api.rs

```diff
@@ -16,6 +16,10 @@ use subspace_runtime_primitives::Moment;
 
 sp_api::decl_runtime_apis! {
     /// Base API that every domain runtime must implement.
+    #[api_version(2)]
+    // `allow(clippy::ptr_arg` is needed because Clippy complains to replace `&Vec<T>` with `&[T]`
+    // but the latter fails to compile.
+    #[allow(clippy::ptr_arg)]
```

Comment:
This fixes the compilation issue, but that issue occurs because the `decl_runtime_apis` macro dereferences the types of its arguments.

So  `&Vec<Block::Extrinsic>` gets turned into `Vec<Block::Extrinsic>` anyway. I'd suggest we just pass an owned vector as an argument, like we do with the other APIs, to make the code more readable.

Reply:
> An owned vector will also require extra cloning, which is what I want to avoid.

I'm not sure if using a reference will avoid any cloning, it might just be hiding it inside the macro. The macro is dereferencing its arguments and causing the type error, so I'm pretty sure it's cloning them somewhere. (But it could be a trait bound, I guess.)

I'll have a look at the macro expansion to check.

Reply:
Looks like it's only cloning in the error case, so this is fine as it is:
<img width="879" alt="Screenshot 2025-05-13 at 09 04 43" src="https://github.com/user-attachments/assets/0908dbd6-de7b-47e1-9456-c60dbc2f1498" />
<img width="766" alt="Screenshot 2025-05-13 at 09 05 08" src="https://github.com/user-attachments/assets/d7e24246-24f9-4b9d-a89f-6d6ee9765868" />

Reply:
I mean the cloning in the caller side, if we use `Vec<Block::Extrinsic>`, then we need to call it with `stateless_runtime_api.decode_extrinsics_prefix(extrinsics.clone())`